torch/_inductor/kernel/mm_common.py

```diff
@@ -339,26 +339,32 @@ def filtered_configs(
     if config["cond"]
 )
 
-# On ROCm convert num_stages to 0 to enable software pipelining
+# On ROCm convert num_stages to 0 or 2 to enable software pipelining
 if torch.version.hip:
+    # on Triton version 3.2.0+, num_stages should be set to 2 instead of 0
+    # https://github.com/triton-lang/triton/blob/627ebbb9843a901dc7e1e88cc48f245d847a3057/third_party/amd/backend/compiler.py#L219
+    import triton
```

Comment:
do we need to check if triton is available?

also, is it possible that this will cause cuda to be initialized before cuda gets used?

https://github.com/pytorch/pytorch/pull/138230#issuecomment-2423354052

^^ in this case, a triton import I added resulted in cuda initialization tests timing out. But I'm not sure this would actually be caught by CI if this was the case, because IDK if we have distributed ROCm tests.

Reply:
There is already triton import here:

https://github.com/pytorch/pytorch/blob/d031d1bf4cf00648ce1fba45ae2561a77918f21d/torch/_inductor/kernel/mm_common.py#L23-L24

This gets folded into `*mm_config` callables which get called as early as in inductor lowering of the GEMMs. My understanding is that the modified file gets imported by the GEMM lowering code, which should be around the same time in the e2e compilation. Your PR has changed dynamo code, which is early on (and not necessarily will invoke inductor). So I feel it's fine (as we'll need to check something *inside* Triton code to gate based on that). But if you have better ideas, I'd be happy to try them out.